src/Microsoft.ML.GenAI.Core/Extension/CausalMLPipelineExtension.cs

```diff
@@ -0,0 +1,51 @@
+﻿// Licensed to the .NET Foundation under one or more agreements.
+// The .NET Foundation licenses this file to you under the MIT license.
+// See the LICENSE file in the project root for more information.
+
+using System;
+using System.Collections.Generic;
+using System.Linq;
+using System.Text;
+using System.Threading.Tasks;
+using static TorchSharp.torch;
+using TorchSharp;
+
+namespace Microsoft.ML.GenAI.Core.Extension;
+
+public static class CausalMLPipelineExtension
+{
+
+    public static string? Generate(
+        this CausalLMPipeline pipeline,
+        string prompt,
+        int maxLen = 128,
+        float temperature = 0.7f,
+        float topP = 0.9f,
+        string[]? stopSequences = null,
+        int eosId = 0,
+        string device = "cpu",
+        bool bos = true,
```

Comment:
We prefer to have these things in a "default" class. If you look at the normal transformers you can see how we do it. 
NOT a blocking issue though.